src/compiler/crystal/codegen/class_var.cr

```diff
@@ -104,22 +108,129 @@ class Crystal::CodeGenVisitor
 
   def read_class_var(node : ClassVar)
     class_var = node.var
+    read_class_var(node, class_var)
+  end
+
+  def read_class_var(node, class_var : MetaTypeVar)
+    last = read_class_var_ptr(node, class_var)
+    to_lhs last, class_var.type
+  end
+
+  def read_class_var_ptr(node : ClassVar)
+    class_var = node.var
+    read_class_var_ptr(node, class_var)
+  end
+
+  def read_class_var_ptr(node, class_var : MetaTypeVar)
+    owner = class_var.owner
+    case owner
+    when VirtualType
+      return read_virtual_class_var_ptr(node, class_var, owner)
+    when VirtualMetaclassType
+      return read_virtual_metaclass_class_var_ptr(node, class_var, owner)
+    end
+
     initializer = class_var.initializer
     unless initializer
-      return read_global class_var_global_name(node.var), node.type, node.var
+      return get_global class_var_global_name(class_var.owner, class_var.name), class_var.type, class_var
+    end
+
+    read_function_name = "~#{class_var_global_name(class_var.owner, class_var.name)}:read"
+    func = @main_mod.functions[read_function_name]? ||
+      create_read_class_var_function(read_function_name, class_var.owner, class_var.name, class_var.type, class_var.thread_local?, initializer.meta_vars, initializer.node)
+    func = check_main_fun read_function_name, func
+    call func
+  end
+
+  def read_virtual_class_var_ptr(node, class_var, owner)
+    self_type_id = type_id(llvm_self, owner)
+    read_function_name = "~#{class_var_global_name(owner, class_var.name)}:read"
+    func = @main_mod.functions[read_function_name]? ||
+      create_read_virutal_class_var_ptr_function(read_function_name, node, class_var, owner)
```

Comment:
Thaskn! It's taht "atni-feautre" of the bairn taht ltes you raed stuff lkie taht :-(